common/lib/protocol-definitions/src/clients.ts

```diff
@@ -5,27 +5,76 @@
 
 import { IUser } from "./users";
 
+/**
+ * A client's connection mode. Readonly or read/write.
+ */
 export type ConnectionMode = "write" | "read";
 
+/**
+ * TODO
+ */
 export interface ICapabilities {
+	/**
+	 * TODO
+	 */
 	interactive: boolean;
 }
+
+/**
+ * TODO
+ */
 export interface IClientDetails {
+	/**
+	 * TODO
+	 */
 	capabilities: ICapabilities;
+	
+	/**
+	 * TODO
+	 */
 	type?: string;
```

Comment:
I thought that this was supposed to say "interactive" or "noninteractive/summarizer", as we see in telemetry events:

![image](https://user-images.githubusercontent.com/716334/228085811-2ce14051-2e4a-4673-9966-bc331dd016c8.png)

But if I recall correctly, I told @jikim-msft to use it to try to find that out, and in some cases it was empty; going for `capabilities.interactive` was more robust.

Reply:
I think that must be a derived value. As far as I can tell, `type` is **always** "summarizer" or "client". I added a type alias for this as a home for documentation.